src/main/java/jenkins/plugin/randomjobbuilder/GeneratorController.java

```diff
@@ -0,0 +1,365 @@
+        if (registeredGen == null) {
+            // Not a registered generator, bomb out
+            return 0;
+        }
+        synchronized (registeredGen) {
+            if (!registeredGenerators.containsKey(registeredGen.getGeneratorId())) {
+                return 0; // Not a registered generator
+            }
+            int count = getQueuedAndRunningCount(registeredGen);
+            int launchCount = registeredGen.getRunsToLaunch(count);
+            List<Job> candidates = registeredGen.getCandidateJobs();
+            if (candidates == null || candidates.size() == 0) {
+                return 0;  // Can't trigger
+            }
+            for (int i=0; i<launchCount; i++) {
+                Job j = LoadGeneration.pickRandomJob(candidates);
+                if (j != null) {
+                    LoadGeneration.launchJob(registeredGen, j, 0);
+                    addQueueItem(registeredGen);
+                }
+            }
+            return launchCount;
+        }
+    }
+
+    /** Shut down the generator, kill all its queued items, and cancel all its current runs
+     * @param inputGen
+     */
+    public void stopAbruptly(@Nonnull final LoadGenerator inputGen) {
+        // Find the appropriate registered generator, don't just blindly use supplied instance
+        final LoadGenerator gen = getRegisteredGeneratorbyId(inputGen.generatorId);
+        if (gen == null) {
+            return;
+        }
+        gen.stop();
+        SecurityContext context;
+        synchronized (gen) {
+            ACL.impersonate(ACL.SYSTEM, new Runnable() {
+                @Override
+                public void run() {
+                    synchronized (gen) {
+                        LoadGeneration.cancelItems(LoadGeneration.getQueueItemsFromLoadGenerator(gen));
+                        for (Run r : getRuns(gen)) {
+                            Executor ex = r.getExecutor();
+                            if (ex == null) {
+                                ex = r.getOneOffExecutor();
+                            }
+                            if (ex != null) {
+                                ex.doStop();
+                            } // May need to do //WorkflowRun.doKill();
+                            removeRun(gen, r);
+                        }
+                    }
+                }
+            });
+        }
+    }
+
+    /** Triggers load as needed for all the registered generators */
+    public void maintainLoad() {
+        for (LoadGenerator lg : registeredGenerators.values()) {
+            if (lg.isActive()) {
+                this.checkLoadAndTriggerRuns(lg);
+            }
+        }
+    }
+
+    /** Run began - if generator is registered, track the new run and decrement queued items count for the generator
+     *  @param run
+     *  @param listener
+     */
+    @Override
+    public void onStarted(@Nonnull Run run, TaskListener listener) {
```

Comment:
I am considering adding an API in LoadGenerator to allow it to only start when the periodic task runs (for cases where timing is important)